### DISCUSSION TITLE
Add Support for Trackpad Gestures

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#827] [Ideas]
### What feature would you like to see?

Something I found to be very convenient from the Arc Browser is the ability to switch Workspaces by two-finger swipes inside the vertical tab bar region using the trackpad. This allows switching between Workspaces to be quicker and more intuitive on laptops. 

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/471_

**Comment by jonaharagon** [comment]
Arc has a lot of trackpad gestures that are really useful. Besides this one, another one I immediately found myself missing in Zen is how when a link opens in a new tab you can do the swipe left (go back one page) gesture, and it closes the tab and goes back to the tab that originally opened that link.

**Comment by baermathias** [comment]
I miss swiping left and right on the (opened) sidebar when switching between workspaces. 
With Zen you have to go to the top with the mouse, select a tab, then go down again and switch to another workspace.

**Reply by ruzfyi** [reply]
I'm confused, can't you do that already? Or are you describing a different action? When I hover over the sidebar I can do a two finger swipe left and right to change workspace accordingly.

**Reply by ruzfyi** [reply]
Oh my God I just realize how old this thread is

**Comment by 23426356587** [comment]
Not sure if this is needed to be said but support for these in mac would be great in addition to windows

**Comment by theherk** [comment]
Works well enough to use shortcuts like in Arc. I use ctrl + number, but I'd really like the swipe action back.

**Comment by raphaelbarreiros** [comment]
On Arc I used to hover the side panel and click Mouse 3 and Mouse 4 to move to the next or previous workspace. That works amazing for me and would love to have this on Zen

**Comment by IrealiTY** [comment]
After swapping from Arc I instinctivel move my cursor to the sidebar and try to swipe to other profiles and nothing happens :') but yeah, that'd be a great feature to have

**Comment by Epoch5427** [comment]
Seeing this feature implemented in Zen Browser would once and for all crown it as the best browser currently available, at least for me. Trackpad gesture support is the only feature that I would say Zen Browser is currently _lacking_

**Comment by gordoncheong** [comment]
Side swipe to switch workspaces works now!

https://github.com/user-attachments/assets/a1715a1d-fde5-4bda-bfe3-d561d288338e

**Reply by theherk** [reply]
It is added which is nice. But at least for me... it doesn't work very well. It is a bit laggy and unreliable. Jumps around a bit.

**Reply by baermathias** [reply]
Doesn't work well. If you have multiple workspaces and you swipe to the side to go to the next workspace it jumps over multiple workspaces and lands not on the next one but on the third or fourth.

**Reply by hoangdinhini** [reply]
Yeah it is way too sensitive, one swipe can switch workspace 1 > 2 > 1. Not really usable for me atm, I am using keyboard shortcuts as a replacement

**Reply by JSGRD22** [reply]
I thought this was fixed since beta release?

**Comment by lguyaux** [comment]
Being able to move picture in picture (PiP) videos with trackpad would be great too

**Reply by JSGRD22** [reply]
You can move them by clicking and dragging anywhere, or am I misinterpreting?

**Reply by lguyaux** [reply]
Indeed, but move it with two fingers just like in Arc feels very intuitive

**Reply by rajdtta** [reply]
Also being able to pinch in/out to change the PiP window size is quite useful

**Comment by lolipodass** [comment]
@mauro-balades i think you can close this now

**Comment by george-emerald** [comment]
This doesn't work on macOS. Where is the feature parity?

**Reply by thebiblelover7** [reply]
It has been working well for me on macOS. Have you updated to the latest version?

**Reply by theherk** [reply]
Has worked on MacOS for a long time.

**Reply by Loskir** [reply]
<img width="487" height="283" alt="image" src="https://github.com/user-attachments/assets/4632e085-42c0-4950-9c0c-70ece4294e2f" />

If it doesn't work for you, check if you have this setting enabled in macos settings. Currently it doesn't work without it for some reason

**Reply by sghng** [reply]
This should definitely be put into the documentation somewhere

**Reply by argumemnon** [reply]
> If it doesn't work for you, check if you have this setting enabled in macos settings. Currently it doesn't work without it for some reason

It shouldn't have to be enabled for this to work. Not a requirement on Arc browser.

**Reply by mohammad-malik** [reply]
They're using spaces as pages code-wise. So when you have that setting not set to two fingers, it doesnt work because its not interpreted as a page swipe. They should have had it be implemented as horizontal scrollbars, which is what arc does afaik. This also explains why it is so much smoother on Arc compared to Zen

**Reply by jonathan-grs** [reply]
I second @mohammad-malik comment. Personally I do not like swipe page navigation, due to risk of context being lost just because I'm grabbing my coffee, so I turn it off in MacOS settings, but that turns off workspace swipe navigation as well. Would be glad if workspace swipe would be decoupled from the MacOS page swipes.
